.github/workflows/ci.yml

```diff
@@ -211,15 +211,14 @@ jobs:
     - name: Set Android Linker path
       if: endsWith(matrix.thing, '-android')
       run: echo "CARGO_TARGET_$(echo ${{ matrix.target }} | tr \\-a-z _A-Z)_LINKER=$ANDROID_NDK/toolchains/llvm/prebuilt/linux-x86_64/bin/$(echo ${{ matrix.target }} | sed s/armv7/armv7a/)21-clang++" >> "$GITHUB_ENV"
-    - if: "!matrix.check_only"
-      run: cargo test --target ${{ matrix.target }} ${{ matrix.extra_test_args }}
-      name: Run tests
+    - name: Build tests
+      # We `build` because we want the linker to verify we are cross-compiling correctly for check-only targets.
+      run: cargo build --target ${{ matrix.target }} --tests ${{ matrix.extra_test_args }}
       shell: bash
       env: ${{ matrix.custom_env }}
-    - if: matrix.check_only
-      # We `build` because we want the linker to verify we are cross-compiling correctly.
-      run: cargo build --target ${{ matrix.target }} --tests
-      name: Build tests
+    - name: Run tests
+      if: "!matrix.check_only"
+      run: cargo test --target ${{ matrix.target }} ${{ matrix.extra_test_args }}
```

Comment:
`cargo test` always builds if the tests are stale, which they will be if you change `CXX`.

Reply:
`CXX` is unchanged between the two steps, so `Run tests` is pretty fast (e.g. [this run](https://github.com/signalapp/boring/actions/runs/6866495218/job/18676118339#step:11:19))

Reply:
Ah, that comment was on an earlier version where you didn't carry over `custom_env`. What you have seems fine.